csodiaq/identifier/targetedReanalysisFunctions.py

```diff
@@ -0,0 +1,232 @@
+from csodiaq.utils import format_protein_list_to_string
+import pandas as pd
+import numpy as np
+
+
+def calculate_mz_of_heavy_version_of_peptide(peptide, lightMz, z):
+    lightAndHeavyLysKMassDiff = 8.014199
+    lightAndHeavyArgRMassDiff = 10.00827
+
+    numLysK = peptide.count("K")
+    numArgR = peptide.count("R")
+
+    return (
+        lightMz
+        + (numLysK * lightAndHeavyLysKMassDiff) / z
+        + (numArgR * lightAndHeavyArgRMassDiff) / z
+    )
+
+
+def filter_to_only_keep_peptides_with_possibly_heavy_K_or_R_terminal_residue(fdrDf):
+    return fdrDf[
+        fdrDf["peptide"].str.endswith("R") | fdrDf["peptide"].str.endswith("K")
+    ].reset_index(drop=True)
+
+
+def filter_to_only_keep_top_peptides_unique_to_protein(fdrDf, maxPeptidesPerProtein):
+    fdrDf = (
+        fdrDf[fdrDf["uniquePeptide"] == 1]
+        .sort_values("ionCount", ascending=False)
+        .reset_index(drop=True)
+    )
+    return (
+        fdrDf.groupby(["leadingProtein"])
+        .head(maxPeptidesPerProtein)
+        .reset_index(drop=True)
+    )
+
+
+def calculate_mz_of_heavy_isotope_of_each_peptide(fdrDf):
+    return fdrDf.apply(
+        lambda x: calculate_mz_of_heavy_version_of_peptide(
+            x["peptide"], x["MzLIB"], x["zLIB"]
+        ),
+        axis=1,
+    )
+
+
+def make_bin_assignments_for_mz_values(mzValues, binWidth=0.75):
+    bins = np.arange(
+        int(min(mzValues)) - 1, int(max(mzValues) + binWidth * 3) + 1, binWidth * 2
+    )
+    values = np.digitize(mzValues, bins)
+    mzBinValues = bins[values]
+    mzBinValues -= binWidth
+    return mzBinValues
+
+
+def filter_out_peptides_based_on_user_settings(
+    fdrDf, isIncludeHeavyIsotopes, maximumPeptidesPerProtein
+):
+    if isIncludeHeavyIsotopes:
+        fdrDf = (
+            filter_to_only_keep_peptides_with_possibly_heavy_K_or_R_terminal_residue(
+                fdrDf
+            )
+        )
+    if maximumPeptidesPerProtein:
```

Comment:
`maximumPeptidesPerProtein` is a number; it'll only evaluate to `False` if it's equal to `0` (in which case it seems like there should be some error anyway; i.e., trying to identify a protein from 0 peptides).

Reply:
I made a function specifically for explaining the nature of "zero" or "non-zero" values and added it to this if statement for clarity.